### PR TITLE
Incorrect tab order on multiple settings pages #7301

### LINKED ISSUE #7301: Incorrect TabOrder on multiple Settings pages
## Current behaviour

The TabOrder on multiple Settings pages is not correct.

## Expected behaviour

Pressing the TAB key should cycle through all controls sequentially. This is great for accessibility.

## Steps to reproduce

- Go to the "Tools" menu, then "Settings"
- Visit the General page, and hit the TAB key multiple times
- Control should move sequentially through all the settings
- It will not

## Screenshots

GitExtensions - incorrect tabOrder

## Environment

- Git Extensions 3.2.1.6628
- Build bc07663c3c893f1d896520e0bdcfbe35cfe4b729
- Git 2.18.0.windows.1 (recommended: 2.22.0 or later)
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4018.0
- DPI 120dpi (125% scaling)

### PR DESCRIPTION
**Pull request by manishkungwani**
Fixes #7301

## Proposed changes

Corrected and verified that the TabOrder should be correct on the following pages
- "Git extensions"
-  General
-  Appearance
-  Appearance > Color
-  Appearance > Fonts
-  Revision Links
-  Build server integration
-  Scripts
-  Hotkeys
-  Shell extensions
-  Advanced
-  Advanced > Confirmations
-  Detailed
-  Detailed > Browse repository window
-  Detailed > Commit dialog
-  Detailed > Diff Viewer
-  SSH
-  Git
-  Git > Paths
-  Git > Config
-  Git > Advanced
-  Plugins

## Test methodology

- Go to the "Tools" menu, then "Settings"
- Visit the General page, and hit the TAB key multiple times
- Control should move sequentially through all the settings

## Test environment(s)

- Git Extensions 3.2.1.6628
- Build bc07663c3c893f1d896520e0bdcfbe35cfe4b729
- Git 2.18.0.windows.1 (recommended: 2.22.0 or later)
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4018.0
- DPI 120dpi (125% scaling)

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by manishkungwani** [comment]
@RussKie @gerhardol can you pl help. I'm not sure what to do, since the AppVeyor build failed due to build time exceeding the max allowed build time.

> Build execution time has reached the maximum allowed time for your plan (60 minutes).

**Comment by pmiossec** [comment]
>Do we need to rerun the tests or any way to skip any passing tests?

When you have the rights on Appveyor, you could re-run a build. Done. We'll see the results in a few minutes.

File: GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceFontsSettingsPage.Designer.cs [hunk 1]
```diff
@@ -98,6 +98,7 @@ private void InitializeComponent()
             this.label56.Text = "Code font";
             this.label56.TextAlign = System.Drawing.ContentAlignment.MiddleLeft;
             // 
+            this.monospaceFontChangeButton.TabIndex = 3;
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceFontsSettingsPage.Designer.cs]: why put it here?

File: GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceSettingsPage.Designer.cs [hunk 2]
```diff
@@ -457,7 +457,7 @@ private void InitializeComponent()
             0});
             this._NO_TRANSLATE_DaysToCacheImages.Name = "_NO_TRANSLATE_DaysToCacheImages";
             this._NO_TRANSLATE_DaysToCacheImages.Size = new System.Drawing.Size(38, 20);
-            this._NO_TRANSLATE_DaysToCacheImages.TabIndex = 3;
+            this._NO_TRANSLATE_DaysToCacheImages.TabIndex = 2;
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceSettingsPage.Designer.cs]: Avatar provider should be after this but is two tabs later

File: GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceFontsSettingsPage.Designer.cs (hunk 1 above)
Comment (manishkungwani) [reply GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceFontsSettingsPage.Designer.cs]: was moved by VS, will fix.

File: GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceSettingsPage.Designer.cs (hunk 2 above)
Comment (manishkungwani) [reply GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceSettingsPage.Designer.cs]: Fix coming up.

**Comment by manishkungwani** [comment]
> Revisions link tab order?
Wasn't able to finish this.
> Scripts Command-Browse-OnEvent-Icon-Window-Arguments is incorrect
This screen's Tabs are behaving odd, I think I've sent the TabIndex correctly.

Please try to review and merge.

I will open a new issue for Revisions link page TabOrder.

File: GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceSettingsPage.Designer.cs [hunk 3]
```diff
@@ -468,7 +468,7 @@ private void InitializeComponent()
             this.pictureAvatarHelp.Name = "pictureAvatarHelp";
             this.pictureAvatarHelp.Size = new System.Drawing.Size(16, 16);
             this.pictureAvatarHelp.SizeMode = System.Windows.Forms.PictureBoxSizeMode.AutoSize;
-            this.pictureAvatarHelp.TabIndex = 8;
+            this.pictureAvatarHelp.TabIndex = 3;
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceSettingsPage.Designer.cs]: Same index as AvatarProvider

Comment (manishkungwani) [reply GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceSettingsPage.Designer.cs]: The help doesn't have a tab action and there's no TabStop on it, is thus inconsequential.

Comment (gerhardol) [reply GitUI/CommandsDialogs/SettingsDialog/Pages/AppearanceSettingsPage.Designer.cs]: Remove then?

**Comment by RussKie** [comment]
Given that changes do no overlap the PR should be squashed

**Comment by manishkungwani** [comment]
> I ran it locally and the following does appear correct:
> 
> * Appearance > Colors
>   appearance-colors

The color buttons do not have TabStops. If that needs to be added, I think I better file a separate issue and a PR.

> * Scripts
>   Up until "Command" tab order is correct, then it jumps around
>   scripts

The TabIndex sequence is correct on that page, despite that, it misbehaves. Will investigate separate.y

PS: I've rebased, squashed the commits and pushed.

**Comment by RussKie** [comment]
> The color buttons do not have TabStops. If that needs to be added, I think I better file a separate issue and a PR.

Sorry, didn't bother look at the code. Indeed those are labels.

**Comment by RussKie** [comment]
@msftbot merge in 24 hours

**Pull request merged**